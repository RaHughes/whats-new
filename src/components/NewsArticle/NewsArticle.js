import React from 'react';
import './NewsArticle.scss';

const NewsArticle = ({img, headline, description, url}) => {
  
  return(
    <article className='news-article'>
      <img src={img} alt='news'/>
      <h2>{headline}</h2>
      <p>{description}</p>
      <form action={url} method="get" target="_blank">
        <button className="article-button" type="submit">Read Full Article</button>
      </form>
    </article>
  )
}

export default NewsArticle;