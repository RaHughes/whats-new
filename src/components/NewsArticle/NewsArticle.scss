article {
  box-sizing: border-box;
  border: 3px solid black;
  padding: 10px;
  height: 50vh;
  width: 70vh;
  background-color: white;

  p {
    width: 100%;
  }
}

img {
  height: 50%;
  width: 100%;
}

.article-button {
  margin-bottom: 0;
  width: 100%;
  height: 50px;
}