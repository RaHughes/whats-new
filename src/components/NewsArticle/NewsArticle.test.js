import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', ()=> { 
  it('It should match snapshot', () => {
    const wrapper = <NewsArticle 
      img="fruits"
      headline="Bananas" 
      description="Fruit Sucks"
      url="www.sucks.com"
      />;

      expect(wrapper).toMatchSnapshot();
  });
});