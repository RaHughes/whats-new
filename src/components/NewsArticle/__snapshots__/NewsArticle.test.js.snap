// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle It should match snapshot 1`] = `ShallowWrapper {}`;

exports[`NewsArticle renders without crashing 1`] = `ShallowWrapper {}`;
